src/components/FieldGuide.js

```diff
@@ -0,0 +1,297 @@
+    const renderItemRow = (item = {}, icons = [], idx) => {
+      return (
+        <FieldGuideItemRow
+          onPress={() => this.openDetail(item)}
+          key={idx}
+          item={item}
+          icons={icons}
+        />
+      )
+    }
+
+    const fieldGuide =
+      <ScrollView>
+        <View onLayout={(event) => this.setHeight(event.nativeEvent.layout.height) }>
+          { addIndex (map)(
+            (item, idx) => {
+              return renderItemRow(item, icons, idx)
+            },
+            items
+          ) }
+        </View>
+      </ScrollView>
+
+    const itemDetail =
+      <FieldGuideItemDetail
+        item={this.state.selectedItem}
+        icons={icons}
+        heightAnim={this.state.heightAnim}
+        onClose={() => this.closeDetail()}
+        setHeaderHeight={(event) => { this.setState({ headerHeight: event.nativeEvent.layout.height + 70 }) }}
+        setContentHeight={(newHeight) => { this.setHeight(newHeight) }}
+      />
+
+    const showFieldGuideItemDetail = !isEmpty(this.state.selectedItem)
+
+    return (
+      <View style={styles.container}>
+        <Animated.View style={[styles.guideContainer, {height: this.state.heightAnim}]}>
+          { showFieldGuideItemDetail ? itemDetail : fieldGuide }
+          { showFieldGuideItemDetail ? backIcon : null }
+          { closeIcon }
+        </Animated.View>
+        { dragBar }
+      </View>
+    )
+  }
+}
+
+const styles = EStyleSheet.create({
+  container: {
+    position: 'absolute',
+    bottom: 0,
+    left: 0,
+    right: 0,
+  },
+  guideContainer: {
+    backgroundColor: 'white',
+    position: 'absolute',
+    bottom: 0,
+    left: 0,
+    right: 0,
+    height: 0,
+    shadowColor: 'rgba(0, 0, 0, 0.24)',
+    shadowOpacity: 0.8,
+    shadowRadius: 5,
+    shadowOffset: {
+      height: 1
+    },
+  },
+  dragBarContainer: {
+    backgroundColor: 'white',
+    position: 'absolute',
+    left: 0,
+    right: 0,
+    height: 18
+  },
+  dragBar: {
+    height: 12,
+    backgroundColor: '$dragBarColor',
+  },
+  dragBarLineAbsoluteContainer: {
+    backgroundColor: 'transparent',
+    position: 'absolute',
+    top: 0,
+    left: 0,
+    right: 0,
+    height: 18
+  },
+  dragBarLineContainer: {
+    alignSelf: 'center',
+    backgroundColor: '$dragBarColor',
+    paddingTop: 4,
+    paddingBottom: 6,
+    paddingHorizontal: 10,
+    borderRadius: 4,
+  },
+  dragBarLine: {
+    backgroundColor: 'white',
+    alignSelf: 'center',
+    height: 2,
+    width: 13,
+    borderRadius: 2,
+    marginTop: 2,
+  },
+  navIcon: {
+    fontSize: 24,
+    color: '$darkTeal',
+    lineHeight: 24,
+  },
+  back: {
+    position: 'absolute',
+    bottom: -28,
+    left: 10,
+    backgroundColor: 'transparent',
+  },
+  close: {
+    position: 'absolute',
+    bottom: -25,
+    right: 15,
+    backgroundColor: 'transparent',
+  },
+})
+
+FieldGuide.propTypes = {
+  guide: React.PropTypes.shape({
+    items: React.PropTypes.arrayOf(React.PropTypes.shape({
```

Comment:
Saying `arrayOf(React.PropTypes.object)` is probably specific enough here, but this is fine for now.